src/FSharp.Control.TaskSeq.SmokeTests/TestUtils.fs

```diff
@@ -0,0 +1,130 @@
+namespace TaskSeq.Tests
+
+open System
+open System.Threading
+open System.Threading.Tasks
+open System.Diagnostics
+open System.Collections.Generic
+
+open Xunit
+open Xunit.Abstractions
+open FsUnit.Xunit
+
+open FSharp.Control
+
+/// Milliseconds
+[<Measure>]
+type ms
+
+/// Microseconds
+[<Measure>]
+type µs
+
+/// Helpers for short waits, as Task.Delay has about 15ms precision.
+/// Inspired by IoT code: https://github.com/dotnet/iot/pull/235/files
+module DelayHelper =
+
+    /// <summary>
+    /// Delay for at least the specified <paramref name="microseconds"/>.
+    /// </summary>
+    /// <param name="microseconds">The number of microseconds to delay.</param>
+    /// <param name="allowThreadYield">
+    /// True to allow yielding the thread. If this is set to false, on single-proc systems
+    /// this will prevent all other code from running.
+    /// </param>
+    let spinWaitDelay (microseconds: int64<µs>) (allowThreadYield: bool) =
+        let start = Stopwatch.GetTimestamp()
+        let minimumTicks = int64 microseconds * Stopwatch.Frequency / 1_000_000L
+
+        // FIXME: though this is part of official IoT code, the `allowThreadYield` version is extremely slow
+        // slower than would be expected from a simple SpinOnce. Though this may be caused by scenarios with
+        // many tasks at once. Have to investigate. See perf smoke tests.
+        if allowThreadYield then
+            let spinWait = SpinWait()
+
+            while Stopwatch.GetTimestamp() - start < minimumTicks do
+                spinWait.SpinOnce(1)
+
+        else
+            while Stopwatch.GetTimestamp() - start < minimumTicks do
+                Thread.SpinWait(1)
+
+    let delayTask (µsecMin: int64<µs>) (µsecMax: int64<µs>) f = task {
+        let rnd = Random()
```

Comment:
concurrent executions of this will all be seeded the same in practical terms. If you can target net7.0, there is a static global random, I think?

Reply:
That's a good point. And yes, the SmokeTest could certainly target a higher .NET 7.0. I'll update.

Reply:
I'll do the update to .NET 7,0 in a separate PR. I'm not a fan of muddling the PRs, plus it allows me to catch any inconsistencies. Will make this a module-level variable, though, so that the call to `Next` has more meaning.

Reply:
Done